CMakeLists.txt

```diff
@@ -25,4 +25,9 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
 	if(BUILD_TESTING)
```

Comment:
This is `if(BUILD_TESTING)` but this is `option(BUILD_TEST ...` above.

I noticed because we introduce the `option(BUILD_EXAMPLE ...` below. I think we should have the `option`s specified in the same place, and that we probably want `BUILD_TEST`, not `BUILD_TESTING`. (I would actually prefer `BUILD_TESTS` and `BUILD_EXAMPLES`, as plural, but I wouldn't block over _that_.

Reply:
@ethomson `BUILD_TESTING` is actually set up via `include(CTest)`, so it is what CMake itself declares to be the official option. So personally I'd agree that `BUILD_TEST` is saner, but CMake disagrees with both of us :/ 

But yeah, you're right, we still have the old `BUILD_TESTS` in here, which has in fact been introduced via f255c0b (ci: convert to use CMake, 2024-09-05). Let me drop that.